### PR TITLE
Move hook `actionOrderStatusPostUpdate` after stock sync

### LINKED ISSUE #22218: Have actualized quantity after call actionOrderStatusPostUpdate
#### Is your feature request related to a problem?

We actually develop a module who report stock.
The stock in Prestashop has been actualized when you change order status according many criteria(logable state, shipped state, ...)
A hook name actionOrderStatusPostUpdate has been call after the order status changed, to permit to do specific operation.
But we see the call of this hook has make after stock updated.
So all module listen this hook cannot have stock actualized.

#### Describe the solution you'd like

Just move the hook call 2 lines after to permit to have the stock updated.

### PR DESCRIPTION
**Pull request by timactive**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch       | develop
| Description  | Physical & Reserved quantity actualized before call the hook actionOrderStatusPostUpdate<br>We develop a module for a client and i wish to have physical quantity value actualized after hook Strange to have the hook call actionOrderStatusPostUpdate before the process has completed.<br><br>Move this hook after the quantity has updated.<br>Permit to have the quantity reserved, available, physiq  actualized<br>We develop a module for a client and i wish to have physical quantity value actualized after hook Strange to have the hook call actionOrderStatusPostUpdate before the process has completed.
| Type         | improvement
| Category     | CO
| BC breaks    | no
| Deprecations | no
| Fixed ticket | Fixes #22218
| How to test  | ## Requirement<br>- Have a module registered to the hook actionOrderStatusPostUpdate who log physical quantity<br>- A product "test" with quantity equal to 1<br><br>## Make a order<br>- Add the product "test" and buy it<br>## Change the  status of order<br>- Change status to shipped<br>## Test ok<br>In module log the physical quantity is equal to 0``

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22219)

### CODE REVIEW

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Hi and thanks for your contribution, Please take time to properly complete the table in your message :wink:

**Review by matthieu-rolland — COMMENTED** [review]
Thank you for your contribution @timactive !

Code wise the fix is OK, but the product team has been tagged in the issue, we'll wait for their feedback before approving anything ;)

**Comment by timactive** [comment]
@PierreRambaud i update the description in table, i think look good
@matthieu-rolland  ok

Thks

**Comment by Progi1984** [comment]
@timactive Thanks ! Could add a scenario for testing, please ?

**Comment by timactive** [comment]
@Progi1984 Updated. Let me know.

**Comment by Progi1984** [comment]
@timactive Could you provide a module or an example with ps_qualityassurance like in #20207 for testing hooks ?

**Comment by timactive** [comment]
Thanks @Progi1984 nice work, I will do it in my free time.

**Comment by marionf** [comment]
Hello @timactive 

As requested by @MatShir we need some informations about this PR :)

https://github.com/PrestaShop/PrestaShop/issues/22218#issuecomment-759169480

Should we have a hook before and after actually. I can see many use cases to have the need to have before and after. And I'm afraid that it could be considered as BC since this hook has here for a long time, isn't it?
WDYT?

**Comment by matks** [comment]
Hello everyone looks like this PR is now stale.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity".

**If however you are still interested in it, please ping us :) we can reopen.**

Thank you for your contribution.

**Pull request closed without merging**
